ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/AssertTopicExecutor.java

```diff
@@ -59,52 +62,75 @@ public static StatementExecutorResponse execute(
           RETRY_MS,
           RETRY_MS,
           () -> assertTopic(
-              client, statement.getStatement().getTopic(), statement.getStatement().getConfig())
+              client,
+              statement.getStatement().getTopic(),
+              statement.getStatement().getConfig(),
+              statement.getStatement().checkExists())
       );
     } catch (final KsqlException e) {
       throw new KsqlRestException(assertionFailedError(e.getMessage()));
     }
     return StatementExecutorResponse.handled(Optional.of(
-        new AssertTopicEntity(statement.getStatementText(), statement.getStatement().getTopic())));
+        new AssertTopicEntity(
+            statement.getStatementText(),
+            statement.getStatement().getTopic(),
+            statement.getStatement().checkExists())));
   }
 
   private static void assertTopic(
       final KafkaTopicClient client,
       final String topic,
-      final Map<String, Literal> config
+      final Map<String, Literal> config,
+      final boolean assertExists
   ) {
     final boolean topicExists;
     try {
       topicExists = client.isTopicExists(topic);
     } catch (final Exception e) {
-      throw new KsqlException("Cannot check that topic exists: " + e.getMessage());
+      throw new KsqlException("Cannot check topic existence: " + e.getMessage());
     }
 
-    if (topicExists) {
-      final List<TopicPartitionInfo> partitionList = client.describeTopic(topic).partitions();
-      final int partitions = partitionList.size();
-      final int replicas = partitionList.get(0).replicas().size();
-      final List<String> configErrors = new ArrayList<>();
-      config.forEach((k, v) -> {
-        if (k.toLowerCase().equals("partitions")) {
-          if (!configMatches(v.getValue(), partitions)) {
-            configErrors.add(
-                createConfigError(topic, "partitions", v.getValue().toString(), partitions));
-          }
-        } else if (k.toLowerCase().equals("replicas")) {
-          if (!configMatches(v.getValue(), replicas)) {
-            configErrors.add(
-                createConfigError(topic, "replicas", v.getValue().toString(), replicas));
-          }
-        } else {
-          configErrors.add("Cannot assert unknown topic property: " + k);
-        }
-      });
-      if (configErrors.size() > 0) {
-        throw new KsqlException(String.join("\n", configErrors));
+    if (!assertExists) {
+      if (config.size() > 0) {
+        LOG.warn("Will skip topic config check for topic non-existence assertion.");
+      }
+      if (topicExists) {
+        throw new KsqlException("Topic " + topic + " exists");
```

Comment:
I'm 95% sure, but just to make sense, but this logic is instead the call which is retried, so if the topic is being deleted, the assertion can pass, etc, etc, right?  (Still waiting for my coffee to kick in.)

Reply:
Yup, if this fails then the entire `assertTopic` method is retried until the timeout